intake/source/base.py

```diff
@@ -215,6 +222,27 @@ def to_spark(self):
         """
         raise NotImplementedError
 
+    @property
+    def entry(self):
+        if self._entry is None:
+            raise ValueError("Source was not made by a catalog")
+        if self._entry() is None:
+            raise RuntimeError("Tried to recover entry, but it was gone")
+        return self._entry()
+
+    def clone(self, **kwargs):
+        return self.entry(**kwargs)
+
+    def __call__(self, *args, **kwargs):
```

Comment:
I assume this needs to be here for back compatibility?  If so, can we deprecate it in favor of `clone`?

Reply:
Yes, this is how it was done before: `entry() -> default source`, `entry(args) -> specific_source`. It could warn indeed, but this is exactly the conversation I wanted to have: what would feel the simplest for a user.
Also, it should maybe just return `self`, not a copy, if the kwargs match the kwargs self was made with.

Reply:
But that implies there is always a reliably way to determine if the kwargs are "equal" and probably an escape hatch to always return a new instance.  It seems like a lot of complexity to save a few keystrokes in a minority of cases.

Reply:
:+1: for always returning a new instance